### PR TITLE
`browserAction.setIcon` by `path` only for Orion

### PR DESCRIPTION
**Pull request by cyfung1031**
see #1939

### CODE REVIEW

**Review by tophf — CHANGES_REQUESTED** [review]
As the comment for `iconDataCache` says caching is necessary.

**Comment by tophf** [comment]
Here's the comment:

> Caching own icon to improve dashboard loading speed, as well as browserAction API
> (e.g. Chrome wastes 40ms in our extension's process to read 4 icons for every tab).

**Comment by cyfung1031** [comment]
> Here's the comment:
> 
> > Caching own icon to improve dashboard loading speed, as well as browserAction API
> > (e.g. Chrome wastes 40ms in our extension's process to read 4 icons for every tab).

OK. Can I make a boolean like "isOrion" then bypass the imageData and just pass the filepath for Orion?

**Comment by tophf** [comment]
Ideally, the bug should be reported to Orion makers so if they fix it there'll be no need for kludges.

File: src/common/safe-globals.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ export const hasOwnProperty = safeApply.call.bind(({}).hasOwnProperty);
 export const safeCall = Object.call.bind(Object.call);
 export const IS_APPLIED = 'isApplied';
 export const IS_FIREFOX = 'geckoProfiler' in chrome;
+export const IS_ORION = typeof webkitIndexedDB === 'object' && chrome.contentScripts && !navigator.userAgentData;
```
Comment (tophf) [inline src/common/safe-globals.js:26]: This may change suddenly in th next version of Orion...

**Comment by tophf** [comment]
So they say they'll fix this bug, which means there's no need for IS_ORION, assuming people use the latest version, which should be the case normally.

**Pull request closed without merging**